vsintegration/tests/Salsa/FSharpLanguageServiceTestable.fs

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Salsa
+
+open System
+open System.IO
+open System.Collections.Generic
+open System.Configuration
+open System.Globalization
+open System.Runtime.InteropServices
+open Microsoft.VisualStudio
+open Microsoft.VisualStudio.Shell
+open Microsoft.VisualStudio.Shell.Interop 
+open Microsoft.VisualStudio.TextManager.Interop 
+open Microsoft.VisualStudio.Text
+open Microsoft.VisualStudio.OLE.Interop
+open Microsoft.FSharp.Compiler
+open Microsoft.FSharp.Compiler.SourceCodeServices
+open Microsoft.VisualStudio.FSharp.LanguageService
+
+type internal FSharpLanguageServiceTestable() as this =
+    static let colorizerGuid = new Guid("{A2976312-7D71-4BB4-A5F8-66A08EBF46C8}") // Guid for colorized user data on IVsTextBuffer
+    let mutable checkerContainerOpt : FSharpChecker option = None
+    let mutable artifacts : ProjectSitesAndFiles option = None
+    let mutable serviceProvider : System.IServiceProvider option = None
+    let mutable preferences : LanguagePreferences option = None
+    let mutable documentationBuilder : IDocumentationBuilder option = None
+    let mutable sourceFactory : (IVsTextLines -> IFSharpSource) option = None
+    let mutable dirtyForTypeCheckFiles : Set<string> = Set.empty
+    let mutable isInitialized = false
+    let mutable unhooked = false
+    let getColorizer (view:IVsTextView) = 
+        let buffer = Com.ThrowOnFailure1(view.GetBuffer())
+        this.GetColorizer(buffer)                         
+
+    let bgRequests = new FSharpLanguageServiceBackgroundRequests(getColorizer,(fun () -> this.FSharpChecker),(fun () -> this.ProjectSitesAndFiles),(fun () -> this.ServiceProvider),(fun () -> this.DocumentationBuilder))
+    
+    member this.FSharpChecker = 
+        if this.Unhooked then raise Error.UseOfUnhookedLanguageServiceState
+        if not this.IsInitialized then raise Error.UseOfUninitializedLanguageServiceState
+        checkerContainerOpt.Value
+
+    member this.ServiceProvider = 
+        if this.Unhooked then raise Error.UseOfUnhookedLanguageServiceState
+        if not this.IsInitialized then raise Error.UseOfUninitializedLanguageServiceState
+        serviceProvider.Value 
+
+    member this.ProjectSitesAndFiles = 
+        if not this.IsInitialized then raise Error.UseOfUninitializedLanguageServiceState
+        artifacts.Value
+        
+    member this.Preferences = 
+        if this.Unhooked then raise Error.UseOfUnhookedLanguageServiceState
+        if not this.IsInitialized then raise Error.UseOfUninitializedLanguageServiceState
+        preferences.Value
+        
+    member this.SourceFactory = 
+        if this.Unhooked then raise Error.UseOfUnhookedLanguageServiceState
+        if not this.IsInitialized then raise Error.UseOfUninitializedLanguageServiceState
```

Comment:
This class is basically exposing functionality of the old language service. I'll be deleting that code iteratively as I expose the new features. The whole class is copied as it is from the old language service in order not to break existing tests.